lib/hackney/income/sql_pause_tenancy_gateway.rb

```
@@ -1,10 +1,15 @@
 module Hackney
   module Income
     class SqlPauseTenancyGateway
-      def set_paused_status(tenancy_ref:, status:)
+      def set_paused_until(tenancy_ref:, until_date:)
```

We now have multiple operations that could raise an error but we raise a custom error in one case, but let the library raise the error in another (update!).I _think_ a more common pattern now would be something like
```ruby
begin
  do dangerous thing1
  do dangerous thing2
rescue FirstBadThing
  raise 'bad thing 1 happened'
rescue SecondBadThing
 raise 'bad thing 2 happened'
rescue AnythingElse
  raise 'something entirely unexpected happened'
end```